heartbeat/monitors/wrappers/summarizer/plugmondur.go

```diff
@@ -0,0 +1,82 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package summarizer
+
+import (
+	"time"
+
+	"github.com/elastic/beats/v7/heartbeat/look"
+	"github.com/elastic/beats/v7/libbeat/beat"
+)
+
+// LightweightDurationSumPlugin handles the logic for writing the `monitor.duration.us` field
+// for lightweight monitors.
+type LightweightDurationSumPlugin struct {
+	startedAt *time.Time
+}
+
+func (lwdsp *LightweightDurationSumPlugin) EachEvent(event *beat.Event, _ error) EachEventActions {
+	// Effectively only runs once, on the first event
+	if lwdsp.startedAt == nil {
+		now := time.Now()
+		lwdsp.startedAt = &now
+	}
+	return 0
+}
+
+func (lwdsp *LightweightDurationSumPlugin) OnSummary(event *beat.Event) OnSummaryActions {
+	_, _ = event.PutValue("monitor.duration.us", look.RTTMS(time.Since(*lwdsp.startedAt)))
+	return 0
+}
+
+func (lwdsp *LightweightDurationSumPlugin) OnRetry() {}
+
+// BrowserDurationSumPlugin handles the logic for writing the `monitor.duration.us` field
+// for browser monitors.
+type BrowserDurationSumPlugin struct {
+	startedAt *time.Time
+	endedAt   *time.Time
+}
+
+func (bwdsp *BrowserDurationSumPlugin) EachEvent(event *beat.Event, _ error) EachEventActions {
+	// Effectively only runs once, on the first event
```

Comment:
The comment doesn't seem to make sense much in the context of browser monitors where each type is distinct.